Usage test failed in x64-windows and x64-windows-static triplet:
1. 
```
CMake Error at vcpkg/installed/x64-windows/share/spng/SPNGConfig.cmake:27 (find_dependency):
  Unknown CMake command "find_dependency".
```
This issue could be fixed by adding `include (CMakeFindDependencyMacro)` on the previous line of `find_dependency(ZLIB REQUIRED)` in https://github.com/randy408/libspng/blob/v0.7.4/cmake/Config.cmake.in

2.
```
CMake Error at F:\VSProject\main\CMakeLists.txt:28 (target_link_libraries):
  Target "main" links to:
    spng::spng
  but the target was not found.
```
The issue is that under the x64-windows-static triplet, the generated target name is `spng::spng_static` instead of `spng::spng`.
For fixing this issue, please modify the content of usage file as below:
```
From: target_link_libraries(main PRIVATE spng::spng)
To: target_link_libraries(main PRIVATE $<IF:$<TARGET_EXISTS:spng::spng>,spng::spng,spng::spng_static>) 
```

--- a/ports/libspng/fix-spngconfig-cmake.patch
+++ b/ports/libspng/fix-spngconfig-cmake.patch
@@ -1,0 +1,11 @@
+diff --git a/cmake/Config.cmake.in b/cmake/Config.cmake.in
+index d0123db..5756179 100644
+--- a/cmake/Config.cmake.in
++++ b/cmake/Config.cmake.in
+@@ -1,5 +1,6 @@
+ @PACKAGE_INIT@
+ 
++include(CMakeFindDependencyMacro)
+ find_dependency(ZLIB REQUIRED)
+ 
+ include("${CMAKE_CURRENT_LIST_DIR}/@targets_export_name@.cmake")

--- a/ports/libspng/portfile.cmake
+++ b/ports/libspng/portfile.cmake
@@ -4,6 +4,8 @@ vcpkg_from_github(
     REF "v${VERSION}"
     SHA512 cd729653599ed97f80d19f3048c1b3bc2ac16f922b3465804b1913bc45d9fc8b28b56bc2121fda36e9d3dcdd12612cced5383313b722a5342b613f8781879f1a
     HEAD_REF master
+    PATCHES
+        fix-spngconfig-cmake.patch
 )
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" SPNG_BUILD_STATIC)

--- a/ports/libspng/usage
+++ b/ports/libspng/usage
@@ -1,4 +1,4 @@
 libspng provides CMake targets:
 
-    find_package(spng CONFIG REQUIRED)
-    target_link_libraries(main PRIVATE spng::spng)
+    find_package(SPNG CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE $<IF:$<TARGET_EXISTS:spng::spng>,spng::spng,spng::spng_static>) 

--- a/versions/l-/libspng.json
+++ b/versions/l-/libspng.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "8efb5a57f5115864001d6eddad855e2958f14fee",
+      "git-tree": "ed674df56d8b39d76443c7db3ae893c08394bf9c",
       "version": "0.7.4",
       "port-version": 0
     }

